waku/waku_core/time.nim

```diff
@@ -9,8 +9,8 @@ import
 
 type Timestamp* = int64 # A nanosecond precision timestamp
 
-proc getNanosecondTime*[T: SomeNumber](timeInSeconds: T): Timestamp =
-  var ns = Timestamp(timeInSeconds * 1_000_000_000.T)
+proc getNanosecondTime*(timeInSeconds: int64 | float64): Timestamp =
+  let ns = Timestamp(timeInSeconds * 1_000_000_000)
   return ns
```

Comment:
Looking at the previous version of the diff (https://github.com/waku-org/nwaku/commit/9bb2f3bbd55f4b1ce738d099ecd49be7bfb8e3a9), there were 12 instances of `getNanosecondTime(getTime().toUnixFloat())`, which I think can be better served by the following function:

```nim
proc getNowInNanosecondTime*(): Timestamp =
  ## Returns the current time with nanosecond precision and range of
  ## 1970 +/- 200 years.

  let now = getTime()
  #            ([-6.4e18 .. -6.4e18]   + [ 0 .. 10^9 ]) ⊂ [-6.5e18 .. -6.5e18]
  #             [-6.5e18 .. -6.5e18] ⊂ [ -9.2e9 .. 9.2e9 ]
  #             [ -9.2e9 .. -9.2e9 ] ⊂ [ -2^63  .. 2^63  ]
  # which fits in [ -9.2e9 .. -9.2e9 ]
  let ns = now.seconds * 1_000_000_000 + now.nanoseconds
  return ns
```

Reply:
Well, this seems reasonable. I am tempted to use `getNowInNanosecondTime` instead of `getNanosecondTime(getTime().toUnixFloat())`